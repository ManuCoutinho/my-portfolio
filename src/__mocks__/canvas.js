HTMLCanvasElement.prototype.getContext = () => {
  return {
    fillRect: jest.fn()
  }
}
