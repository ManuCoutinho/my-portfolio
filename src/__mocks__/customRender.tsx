import { ReactElement, FC, ReactNode } from 'react'
import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
