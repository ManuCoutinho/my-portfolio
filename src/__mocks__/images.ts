export const mockImg = [
	{
		imgPng: '/img1.png',
		imgWebp: '/img1.webp',
		alt: 'img1',
		id: 'img1'
	},
	{
		imgPng: '/img2.png',
		imgWebp: '/img2.webp',
		alt: 'img2',
		id: 'img2'
	}
]
