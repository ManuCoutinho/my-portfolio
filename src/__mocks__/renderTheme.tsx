import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
