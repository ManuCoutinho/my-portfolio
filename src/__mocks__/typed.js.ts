/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-empty-function */

export default class Typed {
  constructor(elementId: string, options: any) {}
  toggle(): void {}
  stop(): void {}
  start(): void {}
  destroy(): void {}
  reset(restart?: boolean): void {}
}
