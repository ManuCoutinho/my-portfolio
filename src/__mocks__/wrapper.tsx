/* eslint-disable react/display-name */
import { ThemeContextProvider } from 'contexts/ThemeContext'
import { DataProviderMock } from '__mocks__/providers/DataContext'

export const customWrapper = () => {
	return ({ children }: { children: React.ReactNode }) => (
		<ThemeContextProvider>
			<DataProviderMock>{children}</DataProviderMock>
		</ThemeContextProvider>
	)
}
