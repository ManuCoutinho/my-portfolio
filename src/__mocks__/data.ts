import { mockImg } from './images'
export default {
  description: 'a beutiful test',
  site: 'http://www.somesit.com',
  repo: 'http://www.somerepo.com',
  styles: 'styles',
  api: 'mock',
  framework: 'next',
  name: 'project',
  tools: ['tools'],
  id: 'id',
  doc: 'http://www.storybook.com',
  coverDefaultLight: 'img.png',
  coverDefaultDark: '',
  coverAltLight: '',
  coverAltDark: '',
  alt: 'mock img',
  img: mockImg,
  route: '/test'
}
