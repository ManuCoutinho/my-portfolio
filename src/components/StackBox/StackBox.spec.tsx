import { screen } from '@testing-library/react'
import { render } from '__mocks__/customRender'
import { StackBox } from '.'

describe('<StackBox/>', () => {
	it('should render a StackBox with as props correctly', () => {
		const att = 'section'
		const { container } = render(<StackBox as={att} />)
		expect(screen.getByTestId('stackbox')).toBeInTheDocument()
		expect(document.getElementsByTagName('section')[0]).toHaveAttribute(
			'data-testid',
			'stackbox'
		)
		expect(container).toMatchInlineSnapshot(`
		<div>
		  <section
		    class="sc-ce9dac95-0 hjZBZn"
		    data-testid="stackbox"
		  />
		</div>
	`)
	})
})
