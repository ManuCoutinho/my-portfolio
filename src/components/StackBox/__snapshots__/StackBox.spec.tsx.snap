// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StackBox/> should render a StackBox component correctly 1`] = `
<div
  class="sc-9a5cd56e-0 loufTH"
  data-testid="stackbox"
>
  Text
</div>
`;
